subprojects/model-core/src/test/groovy/org/gradle/model/internal/core/ModelTypeTest.groovy

```diff
@@ -532,4 +534,32 @@ class ModelTypeTest extends Specification {
         !new ModelType<List<String>>() {}.rawClassOfParameterizedType
         !new ModelType<List<? super String>>() {}.typeVariables[0].rawClassOfParameterizedType
     }
+
+    enum MyEnum {
+        ONE, TWO
+    }
+
+    def "represents enums"() {
+        given:
+        def enumType = MyEnum.class
+        def enumInstanceType = MyEnum.ONE.class
+
+        expect:
+        def modelOfEnum = ModelType.of(enumType)
+        def modelOfInstance = ModelType.of(enumInstanceType)
+        modelOfEnum == modelOfInstance
+    }
+
+    def "represents enums having constants with class body"() {
+        given:
+        def enumType = TimeUnit.class
```

Comment:
:x: It would be better to define an enum type for the test, both for clarity but also for future proofness.

Reply:
Very good point 👍 
I added more coverage and fixed one bug with representation of nested enum constant with class body along the way.